src/main/java/org/openstreetmap/atlas/geography/sharding/AppendCountsForAllZoom.java

```diff
@@ -0,0 +1,179 @@
+package org.openstreetmap.atlas.geography.sharding;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.HashMap;
+
+import org.openstreetmap.atlas.streaming.Streams;
+import org.openstreetmap.atlas.streaming.resource.File;
+import org.openstreetmap.atlas.streaming.resource.WritableResource;
+import org.openstreetmap.atlas.utilities.collections.StringList;
+import org.openstreetmap.atlas.utilities.runtime.Command;
+import org.openstreetmap.atlas.utilities.runtime.CommandMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Command that takes a file with feature counts for each {@link SlippyTile} at a certain zoom
+ * level, and then calculates summed counts for every lower zoom layer to the file. This is meant to
+ * be used as a pre-processing step before creating the sharding tree in
+ * {@link DynamicTileSharding}.
+ *
+ * @author james-gage
+ * @author lcram
+ */
+public class AppendCountsForAllZoom extends Command
+{
+
+    public static final Switch<WritableResource> DEFINITION = new Switch<>("definition",
+            "Resource containing tile to feature count mapping for zoom 0 to maxZoom-1.", File::new,
+            Optionality.REQUIRED);
+    public static final Switch<WritableResource> OUTPUT = new Switch<>("output",
+            "The resource where to save the feature counts for all zooms.", File::new,
+            Optionality.REQUIRED);
+    private static final Switch<Integer> ZOOM = new Switch<>("zoom",
+            "The zoom layer the input csv represents", Integer::valueOf, Optionality.REQUIRED);
+
+    private static final Logger logger = LoggerFactory.getLogger(AppendCountsForAllZoom.class);
+    private static final int READER_REPORT_FREQUENCY = 10_000_000;
+
+    public static void main(final String[] args)
+    {
+        new AppendCountsForAllZoom().run(args);
+    }
+
+    @Override
+    protected int onRun(final CommandMap command)
+    {
+        final Integer zoom = (Integer) command.get(ZOOM);
+        final WritableResource definition = (WritableResource) command.get(DEFINITION);
+        final WritableResource output = (WritableResource) command.get(OUTPUT);
+        final BufferedWriter bufferedWriter = output.writer();
+
+        final int numberLines = linesInFile(definition);
+        HashMap<SlippyTile, Long> countsAtZoom = new HashMap<>(numberLines);
+        int counter = 0;
+        for (final String line : definition.lines())
+        {
+            final StringList split = StringList.split(line, ",");
+            final SlippyTile tile = SlippyTile.forName(split.get(0));
+            countsAtZoom.put(tile, Long.valueOf(split.get(1)));
+            if (++counter % READER_REPORT_FREQUENCY == 0)
+            {
+                logger.info("Read counts for {} zoom level {} tiles.", counter, zoom);
+            }
+        }
+        writeToFile(countsAtZoom, bufferedWriter);
+
+        for (int i = zoom - 1; i >= 0; i--)
```

Comment:
Since we're using `i`inside the loop, can we give it a better name - maybe `zoomLevel`?